lib/granite/action/transaction_manager.rb

```diff
@@ -7,19 +7,24 @@ module TransactionManager
       class << self
         # Runs a block in a transaction
         # It will open a new transaction or append a block to the current one if it exists
-        #
-        # @param [Object] trigger_callbacks_for - object which will receive `run_callbacks(:commit)` after transaction commited
         # @return [Object] result of a block
-        def transaction(trigger_callbacks_for: nil, &block)
-          (callback_listeners << trigger_callbacks_for) if trigger_callbacks_for
-
+        def transaction(&block)
           if in_a_transaction?
             yield
           else
             wrap_in_transaction_with_callbacks(&block)
           end
         end
 
+        # Adds a block or listener object to be executed after finishing the current transaction.
+        # Callbacks are reset after each transaction.
+        # @param [Object] listener an object which will receive `run_callbacks(:commit)` after transaction commited
+        # @param [Proc] block a block whicgh will be called after transaction commited
+        def after_commit(listener = nil, &block)
```

Comment:
The name of the method is confusing I think. At least it confuses me when I saw it [here](https://github.com/toptal/granite/pull/12/files#diff-9bb46447614e76587297ea87589959f9R41) for the first time
Something like "register_listener" will be better I think